integration_tests/tests/test_anomalies_backfill_logic.py

```diff
@@ -0,0 +1,262 @@
+        dateutil.parser.parse(result["bucket_start"]).date(): int(
+            result["metric_value"]
+        )
+        for result in results
+    }
+
+
+def test_full_backfill_for_non_incremental_model(dbt_project: DbtProject, test_id: str):
+    utc_today = datetime.utcnow().date()
+    data_dates = generate_dates(base_date=utc_today - timedelta(1))
+
+    data = [
+        {TIMESTAMP_COLUMN: cur_date.strftime(DATE_FORMAT)}
+        for cur_date in data_dates
+        for _ in range(5)
+    ]
+    test_result = dbt_project.test(
+        test_id, DBT_TEST_NAME, DBT_TEST_ARGS, data=data, as_model=True
+    )
+    assert test_result["status"] == "pass"
+    assert get_row_count_metrics(dbt_project, test_id) == {
+        cur_date: 5
+        for cur_date in data_dates
+        if cur_date >= utc_today - timedelta(DAYS_BACK)
+    }
+
+    data = [
+        {TIMESTAMP_COLUMN: cur_date.strftime(DATE_FORMAT)} for cur_date in data_dates
+    ]
+    test_result = dbt_project.test(
+        test_id, DBT_TEST_NAME, DBT_TEST_ARGS, data=data, as_model=True
+    )
+    assert test_result["status"] == "pass"
+    assert get_row_count_metrics(dbt_project, test_id) == {
+        cur_date: 1
+        for cur_date in data_dates
+        if cur_date >= utc_today - timedelta(DAYS_BACK)
+    }
+
+
+def test_partial_backfill_for_incremental_models(dbt_project: DbtProject, test_id: str):
+    utc_today = datetime.utcnow().date()
+    data_dates = generate_dates(base_date=utc_today - timedelta(1))
+
+    data = [
+        {TIMESTAMP_COLUMN: cur_date.strftime(DATE_FORMAT)}
+        for cur_date in data_dates
+        for _ in range(5)
+    ]
+    test_result = dbt_project.test(
+        test_id,
+        DBT_TEST_NAME,
+        DBT_TEST_ARGS,
+        data=data,
+        as_model=True,
+        materialization="incremental",
+    )
+    assert test_result["status"] == "pass"
+    assert get_row_count_metrics(dbt_project, test_id) == {
+        cur_date: 5
+        for cur_date in data_dates
+        if cur_date >= utc_today - timedelta(DAYS_BACK)
+    }
+
+    # Reload data to the table with 1 row per date instead of 5. If the backfill logic is working,
+    # only metrics for the last 2 days should be updated and the test should fail because the metric
+    # drops from 5 to 1 in these days.
+    data = [
+        {TIMESTAMP_COLUMN: cur_date.strftime(DATE_FORMAT)} for cur_date in data_dates
+    ]
+    test_result = dbt_project.test(
+        test_id,
+        DBT_TEST_NAME,
+        DBT_TEST_ARGS,
+        data=data,
+        as_model=True,
+        materialization="incremental",
+    )
+    assert test_result["status"] == "fail"
+    assert get_row_count_metrics(dbt_project, test_id) == {
+        cur_date: 5 if cur_date < utc_today - timedelta(BACKFILL_DAYS) else 1
+        for cur_date in data_dates
+        if cur_date >= utc_today - timedelta(DAYS_BACK)
+    }
+
+
+def test_longer_backfill_in_case_of_a_gap(dbt_project: DbtProject, test_id: str):
```

Comment:
From this test and below, I ignored the test results and focused on the metrics which are the main goal of this file